src/devices/machine/busmouse.h

```diff
@@ -0,0 +1,74 @@
+// license:BSD-3-Clause
+// copyright-holders:Sergey Svishchev
+/**********************************************************************
+
+	Logitech bus mouse interface emulation
+
+**********************************************************************/
+
+#ifndef MAME_MACHINE_BUSMOUSE_H
+#define MAME_MACHINE_BUSMOUSE_H
+
+#pragma once
+
+
+//**************************************************************************
+//  INTERFACE CONFIGURATION MACROS
+//**************************************************************************
+
+#define MCFG_BUS_MOUSE_EXTINT_CALLBACK(_write) \
+	devcb = &bus_mouse_device::set_extint_wr_callback(*device, DEVCB_##_write);
+
+
+//**************************************************************************
+//  TYPE DEFINITIONS
+//**************************************************************************
+
+// ======================> bus_mouse_device
+
+class bus_mouse_device : public device_t
+{
+public:
+	// construction/destruction
+	bus_mouse_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock);
+
+	template <class Object> static devcb_base &set_extint_wr_callback(device_t &device, Object &&cb) { return downcast<bus_mouse_device &>(device).m_write_extint.set_callback(std::forward<Object>(cb)); }
+
+	// optional information overrides
+	virtual ioport_constructor device_input_ports() const override;
+
+	DECLARE_READ8_MEMBER(ppi_a_r);
+	DECLARE_READ8_MEMBER(ppi_c_r);
+	DECLARE_WRITE8_MEMBER(ppi_c_w);
```

Comment:
There's no need for these handlers and `device_input_ports` to be public (unfortunately the `DECLARE_INPUT_CHANGED_MEMBER` declarations *do* still need to be public).

Reply:
Moved them.